### PR TITLE
basecamp events (schedule entries) api

### PR DESCRIPTION
**Pull request by andrewc5716**
adds create, update, delete functions for basecamp events (schedule entries)

### CODE REVIEW

File: __test__/schedule.test.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        const randomScheduleEntry = getRandomlyGeneratedScheduleEntry();
+        const scheduleIdentifier: ScheduleIdentifier = {
+            projectId: "TEST_PROJECT_ID",
+            scheduleId: "TEST_SCHEDULE_ID"
+        };
+
+        const mockResponse = { id: randomstring.generate() };
+
+        (sendBasecampPostRequest as jest.Mock).mockReturnValue(mockResponse);
+        (getBasecampProjectUrl as jest.Mock).mockReturnValue("https://3.basecamp.com/4474129/buckets/TEST_PROJECT_ID")
```
Comment (jbwong05) [inline __test__/schedule.test.ts:23]: Missing semicolon

File: __test__/schedule.test.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            projectId: randomstring.generate(),
+            scheduleId: randomstring.generate()
+        };
+        const mockResponse = { id: "12345" };
+        (sendBasecampPostRequest as jest.Mock).mockReturnValue(mockResponse);
+        (getBasecampProjectUrl as jest.Mock).mockReturnValue("https://3.basecamp.com/4474129/buckets/TEST_PROJECT_ID")
+
+        // Act
+        const response = createScheduleEntry(randomScheduleEntry, scheduleIdentifier);
+        console.log(response);
```
Comment (jbwong05) [inline __test__/schedule.test.ts:49]: Do we need this log message?

File: __test__/schedule.test.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    it("should return the created schedule entry id", () => {
+        // Arrange
+        const randomScheduleEntry = getRandomlyGeneratedScheduleEntry();
+        const scheduleIdentifier: ScheduleIdentifier = {
+            projectId: randomstring.generate(),
+            scheduleId: randomstring.generate()
+        };
+        const mockResponse = { id: "12345" };
+        (sendBasecampPostRequest as jest.Mock).mockReturnValue(mockResponse);
+        (getBasecampProjectUrl as jest.Mock).mockReturnValue("https://3.basecamp.com/4474129/buckets/TEST_PROJECT_ID")
```
Comment (jbwong05) [inline __test__/schedule.test.ts:45]: Missing semicolon

File: __test__/schedule.test.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        // Arrange
+        const randomScheduleEntry = getRandomlyGeneratedScheduleEntry();
+        const scheduleIdentifier: ScheduleEntryIdentifier = {
+            projectId: "TEST_PROJECT_ID",
+            scheduleEntryId: "TEST_SCHEDULE_ENTRY_ID"
+        };
+
+        const mockResponse = { id: randomstring.generate() };
+        (sendBasecampPutRequest as jest.Mock).mockReturnValue(mockResponse);
+        (getBasecampProjectUrl as jest.Mock).mockReturnValue("https://3.basecamp.com/4474129/buckets/TEST_PROJECT_ID")
```
Comment (jbwong05) [inline __test__/schedule.test.ts:67]: Missing semicolon

File: __test__/schedule.test.ts
Comment (jbwong05) [inline __test__/schedule.test.ts:1]: I know you said yesterday you were having trouble getting the mocking to work so you had to find a workaround. I was able to get the unit tests working in a similar style to some of our existing unit tests. The changes can be seen here - #31. Up to you if you want to adopt them or not

Comment (andrewc5716) [reply __test__/schedule.test.ts:1]: yes, that's what i was looking for, thanks! merged it in

**Review by jbwong05 — APPROVED** [review]
💯

**Pull request merged**